### PR TITLE
pybridge: Ignore read errors from cgroup metrics

### PR DESCRIPTION
**Pull request by martinpitt**
This can fail with "OSError: [Errno 19] No such device" for cgroups that disappear underneath us.

---

Fixes [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19041-20230629-152605-f1fca553-debian-testing-other/log.html#57) -- metrics, the gift that keeps on giving! :wink:

### CODE REVIEW

File: src/cockpit/samples.py [hunk 1]
```diff
@@ -39,6 +42,10 @@ def read_int_file(rootfd: int, statfile: str, default: Optional[int] = None, key
 
     try:
         data = os.read(fd, 1024)
+    except OSError as e:
+        # can run into ENODEV
+        logger.info('Failed to read %s: %s', statfile, e)
```
Comment (jelly) [inline src/cockpit/samples.py:47]: Info, shouldn't this be error?

Comment (martinpitt) [reply src/cockpit/samples.py:47]: We could do this in a more fine-grained manner -- e.g. logger.debug() for ENODEV or ENOENT (expected cases), and logger.warning() for others?

Comment (martinpitt) [reply src/cockpit/samples.py]: ENOENT makes no sense for read(). I adjusted it to be silent for ENODEV, and warn for other errors.

File: src/cockpit/samples.py [hunk 2]
```diff
@@ -39,6 +43,11 @@ def read_int_file(rootfd: int, statfile: str, default: Optional[int] = None, key
 
     try:
         data = os.read(fd, 1024)
+    except OSError as e:
+        # cgroups can disappear between the open and read
+        if e.errno != errno.ENODEV:
+            logger.warning('Failed to read %s: %s', statfile, e)
```
Comment (allisonkarlitskaya) [inline src/cockpit/samples.py:49]: Pushing back in the other direction: if this is really a legit race condition (ie: we expect that groups might disappear sometimes, while scanning) then actually `.warning()` is too much.  `.debug()` makes more sense to me.

Comment (martinpitt) [reply src/cockpit/samples.py:49]: ENODEV is the expected case, which is now quiet. This is more about EIO, EBADF, etc., which are unexpected situations that should cause a test failure IMHO (unexpected message), and also make this possible to spot and debug.

Comment (allisonkarlitskaya) [reply src/cockpit/samples.py:49]: Ah right!  Sorry that I read it wrong.

**Review by allisonkarlitskaya — APPROVED** [review]

**Pull request merged**